### PR TITLE
github graphql api

### PR DESCRIPTION
**Pull request by tanimmahmud**
- Added different component for `repository` and `issue`
- moved types to a different file
- added typescript
- modified created time display to be more human readable
- added react markdown to display issue text
-added style

### CODE REVIEW

File: src/Issue.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import ReactMarkdown from 'react-markdown';
+import './App.css';
+
+type Props = {
+  issue: IssueProps,
+}
+export default class Issue extends Component<Props> {
+  readableDate = (d: string) => {
+    const date = new Date(d);
+    return `${date.getDay()}/${date.getMonth()}/${date.getFullYear()}`;
```
Comment (radar) [inline src/Issue.tsx]: Maybe using the `moment` library for this might be neater?

File: src/Issue.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  }
+  render() {
+    return <div key={this.props.issue.createdAt} className="Issue">
```
Comment (radar) [inline src/Issue.tsx]: Consider using destructuring to get the props out of here.

File: src/Repository.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import { RepositoryProps, IssueProps } from './types'
+
+type Props = {
+  repository: RepositoryProps,
+}
+
+export default class Repository extends Component<Props> {
+  renderIssues = (issues: Array<IssueProps>) => {
+    return issues.map( (issue: IssueProps, i: number) => {
+      return <Issue key={i} issue={issue} />;
```
Comment (radar) [inline src/Repository.tsx]: Use the spread operator here to pass through the issue props: 

```
return <Issue key={i} {...issue}
```

Then you won't need to access each prop with the prefix of `this.props.issue` within the component.

File: src/Repository.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    })
+  }
+  render(){
+    return <div>
+      <h3 className="RepoName"><strong>Repository Name: {this.props.repository.name}</strong></h3>
```
Comment (radar) [inline src/Repository.tsx]: Use destructuring here to get access to the repository props.

File: src/Organization/index.tsx [hunk 5]
```diff
@@ -1,18 +1,23 @@
...
-type OrganizationProps = {
-  login: string,
-  url: string,
-}
+import { OrganizationProps, RepositoryProps } from '../types';
 
 class Organization extends Component<OrganizationProps> {
+  renderRepositories = (repos: Array<RepositoryProps>) => {
+    return repos.map( (repo: RepositoryProps, i: number) => {
+      return <Repository key={i} repository={repo} />;
```
Comment (radar) [inline src/Organization/index.tsx]: Use spread operator here to pass through repository props.

File: src/types.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
+export type AuthorType = {
```
Comment (radar) [inline src/types.tsx]: Good idea moving these out to a separate file.

How does this file handle author being `null`?

File: src/Issue.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    <p><strong>Title:</strong> {this.props.issue.title}</p>
+    <p><strong>Author:</strong>{this.props.issue.author ? this.props.issue.author.login : 'Author Missing'}</p>
+    <div className="IssueText"><strong>Text:</strong><ReactMarkdown source={this.props.issue.bodyText} /></div>
```
Comment (radar) [inline src/Issue.tsx]: I think there is also `bodyHTML` here that the API returns, which is the HTML-ified version of GitHub's markdown.

File: src/Organization/index.tsx [hunk 8]
```diff
@@ -1,18 +1,23 @@
...
+    })
+  }
   render() {
     return (
-      <div>{this.props.login} is available at <a href={`${this.props.url}`}>{this.props.url}</a></div>
+      <React.Fragment>
+        <h1>{this.props.login} is available at <a href={`${this.props.url}`}>{this.props.url}</a></h1>
```
Comment (radar) [inline src/Organization/index.tsx]: Use destructuring here to pull out the props you care about (`login`, `url` + `repositories.nodes`)

File: src/Issue.tsx (hunk 1 above)
Comment (tanimmahmud) [reply src/Issue.tsx]: did not want to add an extra library for this !! but will try with `moment`

File: src/Issue.tsx (hunk 2 above)
Comment (tanimmahmud) [reply src/Issue.tsx]: yep. it is way cleaner and I can use IssueProps type here as well :)

File: src/types.tsx (hunk 6 above)
Comment (tanimmahmud) [reply src/types.tsx]: not handling it here. just rendering 'Author Missing' if author is null from view.

**Pull request closed without merging**